docs/performance/sharding.md

```diff
@@ -1,41 +1,133 @@
+as larger shards can still be processed efficiently by multiple CPU cores during
+query execution.
+Smaller shards also result in reduced Lucene index efficiency, which can adversely
+affect computed search term relevance.
+
+### CPU-to-shard ratio
+
+If most nodes have more shards per table than they have CPUs, the cluster can
+experience performance degradations.
+For example, on clusters with substantial CPU resources (e.g., 8 nodes × 32 CPUs
+= 256 total CPUs), creating 256+ shards per table often proves counterproductive.
+If you don't manually set the number of shards per table, CrateDB will make a
+best guess, based on the assumption that your nodes have two CPUs each.
+
+### 1000 shards per node limit
+
+To avoid _oversharding_, CrateDB by default limits the number of shards per node to
+1_000 as a critical stability boundary. Any operation that would exceed that limit
+leads to an exception.
+For an 8-node cluster, this allows up to 8_000 total shards across all tables.
+Approaching this limit typically indicates a suboptimal sharding strategy rather
+than optimal performance tuning. See also relevant documentation about
+{ref}`table reconfiguration <number-of-shards>` wrt. sharding options.
+
+### Partitions
+
+If you are using {ref}`partitioned tables <crate-reference:partitioned-tables>`,
+note that each partition is clustered into as many shards as you configure
+for the table.
+
+For example, a table with four shards and two partitions will have eight
+shards that can be commonly queried across. But a query that only touches
+one partition will only query across four shards.
+
+How this factors into balancing your shard allocation will depend on the
+types of queries you intend to run.
+
+### Replicas
+
+CrateDB also has replicas of data and this results in additional shards in
+the cluster.
+
+For the purposes of calculating how many shards a table should be clustered
+into, you can typically ignore replica partitions as these are not usually
+queried across for reads.
+
+
+## Caveats
+
+:::{caution}
+:class: hero
+Balancing the number and size of your shards is important for the performance
+and stability of your CrateDB clusters.
 :::
 
-## Optimising for query performance
+### Avoid imbalances
 
-(sharding-under-allocation)=
+- Too many shards can degrade search performance and make the cluster unstable.
+  This is referred to as _oversharding_.
+
+- Very large shards can slow down cluster operations and prolong recovery times
+  after failures.
+
+Finding the right balance when it comes to sharding will vary on a lot of
+things. While it's generally advisable to slightly over-allocate, we
+recommend to benchmark your particular setup to find the sweet spot to
+implement an appropriate sharding strategy.
+
+Figuring out how many shards to use for your tables requires you to think about
+the type of data you're processing, the types of queries you're running, and
+the type of hardware you're using.
```

Comment:
Wrong section? this isn't related to shard imbalance. Shouldn't this rather stay at the general section?

Reply:
Fine. 1d64064f02e6 relocates this to the "General recommendations" section. Afterwards, 882a947bf2 cleans up naming things and layout of the bottom-most section.